linked_list/LinkedList.java

```diff
@@ -0,0 +1,217 @@
+            temp.next = newNode;    
+            sizeOfList++;
+        }
+    }
+    
+    // Method to add an element at a given index
+    public void add(T input, int index) {
+        if (index < 0 || index > sizeOfList)     
+            throw new IndexOutOfBoundsException("Invalid Index");    
+        Node<T> temp = head;
+        if (index == 0)
+            addFront(input);
+        else {
+            for (int i = 0; i < index-1; i++)  
+                temp = temp.next;
+            Node<T> newNode = new Node<T>(input);
+            newNode.next = temp.next;   
+            temp.next = newNode;    
+            sizeOfList++;
+        }
+    }
+    
+    // Method to remove the front element from the Linked List
+    public T removeFront() {
+        if (isEmpty())     
+            throw new NoSuchElementException();
+        T removedElement = head.data; 
+        head = head.next;     
+        sizeOfList--;
+        return removedElement;
+    }
+    
+    // Method to remove the last element from the Linked List
+    public T removeLast() {
+        if (isEmpty())
+            throw new NoSuchElementException();
+        T removedElement;
+        if (head.next == null) {    
+            removedElement = head.data;
+            head = null;
+        }
+        else {    
+            Node<T> temp = head;
+            while (temp.next.next != null)   
+                temp = temp.next;
+            removedElement = temp.next.data;   
+            temp.next = null;  
+        }
+        sizeOfList--;
+        return removedElement;
+    }
+    
+    // Method to remove element at specific index from the Linked List
+    public T remove(int index) {
+        if (isEmpty())
+            throw new NoSuchElementException();
+        if (index < 0 || index > sizeOfList)
+            throw new IndexOutOfBoundsException("Invalid Index");
+        if (index > 0) {
+            Node<T> temp = head;
+            for (int i = 0; i < index-1; i++) 
+                temp = temp.next;       
+            T removedElement = temp.next.data;      
+            temp.next = temp.next.next;    
+            sizeOfList--;
+            return removedElement;
+        }
+        return removeFront();
+    } 
+    
+    // Method to search a element in the Linked List returns its index i
+    public int search(T input) {
+        Node<T> temp = head;
+        for (int i = 0; i < sizeOfList; i++) {
+            if (temp.data == input)
+                return i;   
+            else
+                temp = temp.next;
+        }
+        return -1;     // Will return -1 if not found
+    }
+    
+    // Method to display the Linked List
+    public void display() {
+        Node<T> temp = head;
+        if (temp != null) {
+            System.out.print("[");
+            while (temp != null) {
+                System.out.print(temp.data+", ");
+                temp = temp.next;
+            }
+            System.out.print("\b\b]\n");
```

Comment:
Right. I completely missed the line where opening bracket is printed and therefore thought it to be a typo.